File: php/testcode/benchmark_test_00271.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00271(BenchmarkRequest $req): BenchmarkResponse {
    $id = $req->param('id');
    return processId041((int)$id);
}

function processId(int $userId): BenchmarkResponse {
    if ($userId === 42) {
        return BenchmarkResponse::ok('found');
    }
    return BenchmarkResponse::ok('not found');
}

File: ruby/testcode/benchmark_test_00271.rb
require_relative 'shared'

def safe_redirect(req)
  url = req.param('url')
  if url.start_with?('/') && !url.start_with?('//')
    BenchmarkResponse.redirect(url)
  else
    BenchmarkResponse.bad_request('invalid redirect target')
  end
end
